### PR TITLE
support preemption when the number of pods of a node reaches the upper limit

### PR DESCRIPTION
**Pull request by Lily922**
support feature described in https://github.com/volcano-sh/volcano/issues/3191

### CODE REVIEW

File: pkg/scheduler/cache/event_handlers.go [hunk 1]
```diff
@@ -66,6 +73,126 @@ func (sc *SchedulerCache) getOrCreateJob(pi *schedulingapi.TaskInfo) *scheduling
...
+			pvcName = vol.PersistentVolumeClaim.ClaimName
+		case vol.Ephemeral != nil:
+			// Generic ephemeral inline volumes also use a PVC,
+			// just with a computed name and certain ownership.
+			// That is checked below once the pvc object is
+			// retrieved.
+			pvcName = ephemeral.VolumeClaimName(pod, &vol)
+			isEphemeral = true
+		default:
+			// do not support Inline Volume now.
```
Comment (neujie) [inline pkg/scheduler/cache/event_handlers.go:106]: 可以补充下注释。
// TODO Currently, only CSI PVC volumes are supported. If you want to support non-CSI PVC volumes, you can implement here.

Comment (neujie) [inline pkg/scheduler/cache/event_handlers.go:106]: 可以补充下注释。比如：目前仅支持CSI PVC卷。如果想支持非CSI PVC卷，可以在这里实现。

File: pkg/scheduler/cache/event_handlers.go [hunk 2]
```diff
@@ -66,6 +73,126 @@ func (sc *SchedulerCache) getOrCreateJob(pi *schedulingapi.TaskInfo) *scheduling
 	return sc.Jobs[pi.Job]
 }
 
+func (sc *SchedulerCache) addPodCSIVolumesToTask(pi *schedulingapi.TaskInfo) error {
```
Comment (william-wang) [inline pkg/scheduler/cache/event_handlers.go:76]: Suggest seperate CSIVolume in another new PR.

File: pkg/scheduler/api/pod_info.go [hunk 3]
```diff
@@ -65,6 +65,7 @@ func GetPodResourceRequest(pod *v1.Pod) *Resource {
 	for _, container := range pod.Spec.InitContainers {
 		result.SetMaxResource(NewResource(container.Resources.Requests))
 	}
+	result.AddScalar(v1.ResourcePods, 1)
```
Comment (william-wang) [inline pkg/scheduler/api/pod_info.go:68]: what's the intention of add this?

**Comment by william-wang** [comment]
The fix looks good to me overall :)

**Review by william-wang — COMMENTED** [review]
/lgtm

**Review by william-wang — APPROVED** [review]
/approve

**Pull request merged**